app/javascript/components/Profile/UserList.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Fragment } from 'react';
+import { Icon, Table, Button } from 'semantic-ui-react';
+import { inject, observer } from 'mobx-react';
+import { computed } from 'mobx';
+import UserStore, { ReloadState } from '../../stores/user_store';
+import User, { Role } from '../../models/User';
+import { deleteUser, updateUser } from '../../api/admin';
+import SessionStore from '../../stores/session_store';
+
+interface InjectedProps {
+  userStore: UserStore;
+  sessionStore: SessionStore;
+}
+
+@inject('userStore', 'sessionStore')
+@observer
+export default class UserList extends React.Component {
+  componentDidMount() {
+    this.injected.userStore.loadUsers();
+  }
+
+  get injected() {
+    return this.props as InjectedProps;
+  }
+
+  onDeleteUser(id: string) {
+    deleteUser(id).then(() => this.injected.userStore.loadUsers(true));
+  }
+
+  setUserRole(id: string, role: Role) {
+    updateUser(id, { role: role }).then(() => this.injected.userStore.loadUsers(true));
+  }
+
+  @computed get reloadIcon() {
+    const { reloadState } = this.injected.userStore;
+    switch (reloadState) {
+      case ReloadState.Error:
+        return 'times circle';
+      case ReloadState.Success:
+        return 'check circle';
+      default:
+        return 'refresh';
+    }
+  }
+
+  @computed get reloadIconColor() {
+    const { reloadState } = this.injected.userStore;
```

Comment:
this is somehow repeated logic (comapted with reloadIcon). Maybe, this can somehow be generalized. :smiley_cat: